File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/model/RefreshToken.java
package com.ntnu.gidd.model;

import lombok.*;
import org.hibernate.annotations.Type;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import java.util.UUID;


@Data
@ToString(of = {"jti", "isValid"})
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class RefreshToken {
    @Id
    @Type(type="uuid-char")
    private UUID jti;

    private boolean isValid;

    @OneToOne
    private RefreshToken next;

}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/security/token/RefreshToken.kt
package ntnu.idatt2105.security.token

import org.hibernate.annotations.Type
import java.util.*
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToOne

@Entity
data class RefreshToken(
    @Id
    @Type(type = "uuid-char")
    val jti: UUID,

    var isValid: Boolean,

    @OneToOne
    var next: RefreshToken?

)
